ergotree-ir/src/chain/header.rs

```diff
@@ -42,23 +49,211 @@ pub struct Header {
     /// Root hash of extension section
     #[cfg_attr(feature = "json", serde(rename = "extensionHash"))]
     pub extension_root: Digest32,
+    /// Solution for an Autolykos PoW puzzle
+    #[cfg_attr(feature = "json", serde(rename = "powSolutions"))]
+    pub autolykos_solution: AutolykosSolution,
+    /// Miner votes for changing system parameters.
+    /// 3 bytes in accordance to Scala implementation, but will use `Vec` until further improvements
+    #[cfg_attr(feature = "json", serde(rename = "votes"))]
+    pub votes: Votes,
+}
+
+impl Header {
+    /// Used in nipowpow
+    pub fn serialize_without_pow(&self) -> Result<Vec<u8>, ScorexSerializationError> {
+        use byteorder::{BigEndian, WriteBytesExt};
+        let mut data = Vec::new();
+        let mut w = SigmaByteWriter::new(&mut data, None);
+        w.put_u8(self.version)?;
+        self.parent_id.0.scorex_serialize(&mut w)?;
+        self.ad_proofs_root.scorex_serialize(&mut w)?;
+        self.transaction_root.scorex_serialize(&mut w)?;
+        self.state_root.scorex_serialize(&mut w)?;
+        w.put_u64(self.timestamp)?;
+        self.extension_root.scorex_serialize(&mut w)?;
+
+        // n_bits needs to be serialized in big-endian format. Note that it actually fits in a
+        // `u32`.
+        let mut n_bits_writer = vec![];
+        #[allow(clippy::unwrap_used)]
+        n_bits_writer
+            .write_u32::<BigEndian>(self.n_bits as u32)
+            .unwrap();
+        w.write_all(&n_bits_writer)?;
+
+        w.put_u32(self.height)?;
+        w.write_all(&self.votes.0)?;
+
+        // For block version >= 2, this new byte encodes length of possible new fields.
+        // Set to 0 for now, so no new fields.
+        if self.version > 1 {
+            w.put_i8(0)?;
+        }
+        Ok(data)
+    }
+}
+
+impl ScorexSerializable for Header {
```

Comment:
It looks like we have everything in place to do a serialization roundtrip test for `Header`, right? Like in https://github.com/kettlebell/sigma-rust/blob/f7d2452c199d918c8d12f600283636bd02cc7909/ergo-p2p/src/peer_spec.rs#L195-L198

Reply:
Done.